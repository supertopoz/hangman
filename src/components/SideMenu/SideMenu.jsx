import React from "react";
import styled from "styled-components"
import { BrowserRouter as Router, Route, Link } from 'react-router-dom'

const Menu = styled.div`
  display: grid;
  margin-top: -8px;
  margin-left: -10px;
  z-index: 10;
  position: absolute;
  height: 100%;
  width: 100%;
  grid-template-columns: 2fr 10fr;
`

const MenuHeader = styled.div`
    height: 65px;
    background: ghostwhite;
    border-bottom: 1px solid lightgrey;
`

const MenuLink = styled.div`
    margin-left: 10px;
    padding: 10px;
    font-size: 1.5rem;

`

const LeftColumn = styled.div`
  background: white;
  z-index: 200;
  box-shadow: 3px 0px 4px #9E9E9E;

`

const RightColumn = styled.div`
  background: transparent;
`

class SideMenu extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            sideMenuVisibility: false
        }
    }

    hideMenu (){
       this.setState({sideMenuVisibility: false});
    }

    componentDidMount(){
       if(this.props.menu){
         this.setState({sideMenuVisibility: true});
       }
       

    }


    render(){
    if(this.state.sideMenuVisibility){
    return (

        <Menu>
        <LeftColumn onMouseLeave={(e)=> this.hideMenu()}>
        <MenuHeader/>
        <MenuLink>
        <Link to="/about" onClick={() => this.hideMenu()}>About</Link>
        </MenuLink>
        <MenuLink>
        <Link to="/home" onClick={() => this.hideMenu()}>Home</Link>
        </MenuLink>
        </LeftColumn>
        </Menu>
    );
    } 
    return (<div></div>)
    }
};

export default SideMenu;