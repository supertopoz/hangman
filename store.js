import { createStore, combineReducers, applyMiddleware } from "redux";
import { createLogger } from "redux-logger";

import student from "./src/reducers/studentDetailsReducer.js";
import pageAnimations from "./src/reducers/pageAnimations.js";
import user from "./src/reducers/userReducer.js";

export default createStore(
    combineReducers({ student, user, pageAnimations }), {},
    applyMiddleware(createLogger())
);