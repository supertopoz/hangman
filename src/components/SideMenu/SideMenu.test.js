import React from 'react';
import { shallow } from 'enzyme';
// setup file
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import store from "../../../store";

configure({ adapter: new Adapter() });

import { SideMenu } from './SideMenu';

describe('SideMenu', () => {
  const app = shallow(<SideMenu />);
  it('renders the side menu', () => {
    expect(app).toMatchSnapshot();
  });
});