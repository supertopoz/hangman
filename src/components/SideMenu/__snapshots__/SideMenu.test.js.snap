// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu renders the side menu 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs-12"
        >
          <h1>
            Info Page
          </h1>
        </div>,
        <div>
          <div>
            <button
              onClick={[Function]}
            >
              Change the Username
            </button>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Info Page
          </h1>,
          "className": "col-xs-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Info Page",
          },
          "ref": null,
          "rendered": "Info Page",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <button
              onClick={[Function]}
            >
              Change the Username
            </button>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
            >
              Change the Username
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Change the Username",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Change the Username",
            "type": "button",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs-12"
          >
            <h1>
              Info Page
            </h1>
          </div>,
          <div>
            <div>
              <button
                onClick={[Function]}
              >
                Change the Username
              </button>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Info Page
            </h1>,
            "className": "col-xs-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Info Page",
            },
            "ref": null,
            "rendered": "Info Page",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <button
                onClick={[Function]}
              >
                Change the Username
              </button>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
              >
                Change the Username
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Change the Username",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Change the Username",
              "type": "button",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
